examples/loopback.rs

```diff
@@ -117,8 +117,8 @@ fn main() {
 
     let mut socket_set_entries: [_; 2] = Default::default();
     let mut socket_set = SocketSet::new(&mut socket_set_entries[..]);
-    let server_handle = socket_set.add(server_socket);
-    let client_handle = socket_set.add(client_socket);
+    let server_handle = socket_set.add(server_socket.into());
```

Comment:
To avoid this, `SocketSet::add` should have the signature `fn add<T: Into<Socket>>(socket: T)`.

Reply:
Yes, I thought about this, but I wasn't sure if you're okay with the resulting increase in file size. But I assume that it's negligible.

Reply:
It might not be, but space profiling is something that should be done separately and preferably without degrading API ergonomics.